core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -4548,88 +4549,95 @@ protected SqlBuilder getQueryString(boolean isGroupFiltering, boolean isUsername
         SqlBuilder sqlBuilder;
         boolean hitGroupFilter = false;
         boolean hitClaimFilter = false;
-        int groupFilterCount = 0;
-        int claimFilterCount = 0;
+        List<SqlBuilder> mysqlSubSqlBuilders = new ArrayList<>();
+        boolean isGroupFilteringWithNEOperator = isGroupFilteringWithNotEqualOperator(isGroupFiltering, expressionConditions);
+        String groupUserJoin =  isGroupFilteringWithNEOperator ? " RIGHT JOIN " : " INNER JOIN ";
```

Comment:
userRoleAndUserJoin

A proper variable name which indicates the joining tables.

Reply:
Resolved via https://github.com/wso2/carbon-kernel/pull/4305/commits/ac37534bb69427c5149b7558b89fd10120e62ad9